---
title: "Lab 14 Homework"
author: "Please Add Your Name Here"
date: "`r Sys.Date()`"
output: 
  html_document: 
    keep_md: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Instructions
Answer the following questions and complete the exercises in RMarkdown. Please embed all of your code and push your final work to your repository. Your final lab report should be organized, clean, and run free from errors. Remember, you must remove the `#` for the included code chunks to run. Be sure to add your name to the author header above. For any included plots, make sure they are clearly labeled. You are free to use any plot type that you feel best communicates the results of your analysis.  

Make sure to use the formatting conventions of RMarkdown to make your report neat and clean!  

Looping over files for ggplot
```{r}
data_files3 <- list.files("data/us_individual_collar_data", full.names = TRUE)
# summary table
data_list <- lapply(data_files3, read_csv)
(collar_data_all <- bind_rows(data_list))
collar_data_all %>% 
  summarise(max_lat=max(lat),
            min_lat=min(lat),
            max_long=max(long),
            min_long=min(long))
# only ggplot
for (i in 1:length(data_files3)){
  data0 <- as.data.frame(read_csv(data_files3[i]))
  print(
    ggplot(data0, aes(x=long,y=lat,color=time))+
      geom_path()+
      geom_point()+
      labs(title = data_files3[i], x = "longitude", y = "latitude")
  )
}
# on a map
lat <- c(14.91539, 31.56526)
long <- c(122.0874, 134.9906)
bbox <- make_bbox(long, lat, f = 0.5)
map <- get_map(bbox, maptype = "terrain", source = "stamen")
ggmap(map)

for (i in 1:length(data_files3)){
  data0 <- as.data.frame(read_csv(data_files3[i]))
  print(
    ggmap(map)+
      geom_path(data = data0, aes(long,lat))+
      geom_point(data = data0, aes(long,lat,color=time))+
      labs(title = data_files3[i], x = "longitude", y = "latitude")
  )
}
```

## Load the libraries
```{r message=FALSE, warning=FALSE}
library(qtl)
library(qtlcharts)
```

## Load the Data
We will use the data from an experiment on hypertension in the mouse [Sugiyama et al., Genomics 71:70-77, 2001](https://www.sciencedirect.com/science/article/pii/S0888754300964012?via%3Dihub)
```{r}
#?hyper
data(hyper)
```

1. Create a summary of the hypertension data. How many individuals and phenotypes are included in this data set? How many gene markers and chromosomes are included in this data set? Please create a table to show the number of markers on each chromosome.
```{r}
summary(hyper)
nind(hyper)    # No. individuals
nphe(hyper)    # No. phenotypes
nchr(hyper)    # No. chromosomes
totmar(hyper)  # Total markers
nmar(hyper)    # No. markers on each chromosome
```
2. Please make an interactive genetic map of markers for the hypertension data.
```{r}
iplotMap(hyper)
```
3. Make a plot shows the pattern of missing genotype data in the hypertension dataset. Please reorder the recorded individuals according to their blood pressure phenotypes. Is there a specific pattern of missing genotype? Please explain it.
```{r}
plotMissing(hyper, main="")
head(hyper$pheno)
plotMissing(hyper, main="", reorder=1)
```

4. Based on your answer from Q4, you probably noticed that there are gene markers without data. Please use the function drop.nullmarkers to remove markers that have no genotype data. After this, make a new summary to show the number of markers on each chromosome. How many gene markers were dropped? Where were the dropped markers located?
```{r}
totmar(hyper)  # Total markers
nmar(hyper)    # No. markers on each chromosome
hyper <- drop.nullmarkers(hyper)
totmar(hyper)  # Total markers
nmar(hyper)    # No. markers on each chromosome
```
5. Please conduct single-QTL analysis and create a table to give the maximum LOD score on each chromosome based on their blood pressure phenotypes. Which gene marker has the higiest LOD score? Which chromosome contains the gene marker that has the highest LOD score? Then, creates an interactive chart with LOD curves from a genome scan for all chromosomes.
```{r}
hyper <- calc.genoprob(hyper, step=1)
out.em <- scanone(hyper, pheno.col=1)
summary(out.em)
max(out.em)
iplotScanone(out.em)
```

6. Based on your genome scan results, create a table which only includes those chromosomes with LOD > 1. Make a new interactive chart with LOD curves which only show those chromosomes with LOD > 1.
```{r}
summary(out.em, threshold=1)
iplotScanone(out.em, chr=c(1, 2, 4, 5, 6, 15))
```
7. Creates an interactive chart with LOD curves linked to estimated QTL effects for only those chromosomes with LOD > 1. Find the gene maker with the highest LOD score and describe how does the genetype of this marker influence the individual's phenotype. Then, save your interactive chart from Q9 as a html file and make sure your upload it to your github repository with your lab14 homework as well.
```{r}
iplotScanone(out.em, hyper, chr=c(1, 2, 4, 5, 6, 15))
hyper_iplotScanone <- iplotScanone(out.em, hyper, chr=c(1, 2, 4, 5, 6, 15))
htmlwidgets::saveWidget(hyper_iplotScanone, file="hyper_iplotScanone.html")
```

